### PR TITLE
feat(DropdownToggle): Add splitButtonItems prop, DropdownToggleCheckbox and split-button dropdown examples

### LINKED ISSUE #1287: Add split-button dropdown
https://github.com/patternfly/patternfly-next/pull/1263/

This is a new variation of a dropdown. The current dropdown toggle element changed from a `button` to a `div`, and you can now put interactive elements in the toggle `div` and those control the menu. Our current example shows a `button.pf-c-dropdown__toggle-button` and `label.pf-c-dropdown__toggle-check` (with an `<input type="checkbox">` in it) in the toggle area, but that could be 3 buttons, 2 buttons and a check, etc. We may add more things like the `pf-c-dropdown__toggle-button/check` later. The aria properties will move from the current `button.pf-c-dropdown__toggle` to the element(s) that control the menu - in this example, that's `.button.pf-c-dropdown__toggle-button`.

### PR DESCRIPTION
**Pull request by mturley**
Closes #1287

**What**: Adds a new prop `splitButtonItems` to the `DropdownToggle` component which causes it to render a wrapper div surrounding those items and the dropdown toggle button. Also adds a new component `DropdownToggleCheckbox` for use as an element in `splitButtonItems`, which copies all of the props of the `Checkbox` component except `label`, and renders the checkbox as a child of the label element instead of rendering a checkbox with an adjacent label.

Note: I didn't write the new component in TypeScript, just because none of the Dropdown components are in TypeScript yet and I figure we should convert them all at once.

### CODE REVIEW

**Comment by mturley** [comment]
I found a console warning on the Dropdown docs page that appears to have already existed on master. I don't think it should block this PR, in case anyone notices it. Opened an issue for it: https://github.com/patternfly/patternfly-react/issues/1844

**Comment by redallen** [comment]
There are a lot of such warnings @mturley, and in the Jest tests too :)

File: packages/patternfly-4/react-core/src/components/Dropdown/DropdownToggle.js [hunk 1]
```diff
@@ -5,19 +5,38 @@ import Toggle from './Toggle';
...
+      )}
+    >
+      {splitButtonItems}
+      <InnerToggle {...props} isSplitButton ariaLabel={props.ariaLabel || 'Select'} />
+    </div>
+  ) : (
+    <InnerToggle {...props} />
+  );
+
+// TODO add splitButtonItems prop
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/Dropdown/DropdownToggle.js]: Well, you added it.

File: packages/patternfly-4/react-core/src/components/Dropdown/Dropdown.md [hunk 2]
```diff
@@ -349,6 +349,142 @@ class IconDropdown extends React.Component {
...
+      <DropdownItem key="separated action" component="button">
+        Separated Action
+      </DropdownItem>
+    ];
+    return (
+      <Dropdown
+        onSelect={this.onSelect}
+        toggle={(
+          <DropdownToggle
+            splitButtonItems={[
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/Dropdown/Dropdown.md]: Good choice :)

**Comment by christiemolloy** [comment]
@mcarrano @jgiardino when a user clicks on the checkbox after the dropdown toggle has been clicked once, should the dropdown menu remain open?

**Comment by redallen** [comment]
You also have 20 snapshots to update. Use `yarn test:pf4 -u`.

File: packages/patternfly-4/react-core/src/components/Dropdown/DropdownToggle.js (hunk 1 above)
Comment (mturley) [reply packages/patternfly-4/react-core/src/components/Dropdown/DropdownToggle.js]: oops, haha.

**Comment by mturley** [comment]
Thanks @redallen. I also just noticed my handling of `ariaLabel` is a little weird (and caused the aria-label to disappear in a lot of those snapshots), I'm just going to standardize on `aria-label` and fix that.

**Comment by mcarrano** [comment]
@christiemolloy I think you raise a good question. I don't feel strongly either way. It's definitely an edge case and am OK with the way it works now. @jgiardino what do you think?

**Comment by mturley** [comment]
I really don't like how `InnerToggle` is being exposed in the snapshots and making them harder to read. I think I'll refactor slightly to remove the helper component. Feel free to let me know if you guys think I should revert this next commit.

**Comment by mturley** [comment]
There, now the existing snapshots don't need to change at all :)

File: packages/patternfly-4/react-core/src/components/Dropdown/Dropdown.md [hunk 3]
```diff
@@ -349,6 +349,142 @@ class IconDropdown extends React.Component {
...
+              <DropdownToggleCheckbox
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Dropdown/Dropdown.md]: I like that this is flexible for future use cases of the split button as described in the issue.

**Comment by jgiardino** [comment]
I will not get a chance to review this today to confirm, but any instances of aria-label should be exposed to the consumer so that they can define their own text strings. The naming convention we're using is `aria-label` if the component represents a singular element (like Button), or something more specific if there are multiple elements in a component that can have an aria-label, like `ariaLabelCheckbox`.

**Comment by mturley** [comment]
Yep, I made sure of that @jgiardino 😄 . The `aria-label` prop on `DropdownToggle` is optional and defaults to `"Select"` for split button toggles, or defaults to no aria-label for regular toggles since they have label text as a child. In both cases it can be specified by the consumer. The `aria-label` prop on `DropdownToggleCheckbox` is required (must be specified by the consumer), which matches what we were already doing for the regular `Checkbox`.

Good to know about the naming convention. I happened to get it right this time but I'll keep that in mind in the future.

**Comment by jgiardino** [comment]
I never doubted you, @mturley! And thanks for the explanation! 
LGTM

**Pull request merged**